Observer/Product/SaveAfter.php

```diff
@@ -312,4 +298,19 @@ private function unassignProductChildrensForSync($product)
         $this->manageUnAssign($productChildrenIdsBeforeSave, $productChildrenIds, $product);
         $this->catalogSession->unsChildrenIds();
     }
+
+    /**
+     * @param array<int, int>|int $firstArray
+     * @param array<int, int>|int $secondArray
+     * @return array<int, int>
+     */
+    private function findDiffBetweenArrays($firstArray, $secondArray)
```

Comment:
`findDifferenceBetweenArrays`

Reply:
I think `findDiffBetweenArrays` is more accurate, but still not a good name, as it finds the diff and also retrieve it. I'm  
changing it to `findAndRetrieveDiffBetweenArrays` wdyt?
![image](https://user-images.githubusercontent.com/10895971/158849689-e1eaff1d-d13d-4faf-b4e1-97a12621d9be.png)